.github/workflows/ci.yml

```diff
@@ -188,6 +188,44 @@ jobs:
     - name: Check formatting
       run: cargo fmt -- --check
 
+  gather_workspace_members:
+    runs-on: ubuntu-latest
+    outputs:
+      members: ${{ steps.cargo-metadata.outputs.members }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - id: cargo-metadata
+        run: |
+          WORKSPACE_MEMBERS=$(cargo metadata --format-version=1 --no-deps | jq -c '.packages |.[] | .name' | jq -s | jq -c '.')
+          echo "::set-output name=members::${WORKSPACE_MEMBERS}"
+
+  semver-check:
+    runs-on: ubuntu-latest
+    needs: gather_workspace_members
+    strategy:
+      matrix:
+        crate: ${{ fromJSON(needs.gather_workspace_members.outputs.members) }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install rust nightly
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+
+      - run: cargo install cargo-semver-checks
+      
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Semver Check
+        if: ${{ matrix.crate != 'libp2p' && contains(matrix.crate, 'libp2p') }}  
```

Comment:
What exactly is this doing? If we don't want to run the tool on certain crates, it is probably better to exclude them in the `jq` expression above!

Reply:
you're absolutely right, that makes much more sense.